library/assumptions.ml

```diff
@@ -33,10 +36,20 @@ type context_object =
 module OrderedContextObject =
 struct
   type t = context_object
+
+  let compare_axiom x y =
+    match x,y with
+    | Constant k1 , Constant k2 ->
+        con_ord k1 k2
+    | Positive m1 , Positive m2 ->
+        MutInd.CanOrd.compare m1 m2
+    | Positive _ , Constant _ -> 1
+    | _ -> -1
+
```

Comment:
The scalable way to write such comparisons is:

```
| Constant k1, Constant k2 -> ...
| Constant _, _ -> -1
| _, Constant _ -> 1
| Axiom a1, Axiom a2 -> ...
``` 

This has size linear in the number of constructors, and adding an extra constructor always result in an lack-of-exhaustivity warning.